scim-sdk-server/src/main/java/de/captaingoldfish/scim/sdk/server/patch/MsAzurePatchReplaceWorkaroundHandler.java

```diff
@@ -0,0 +1,198 @@
+ * }
+ * </pre>
+ *
+ * the value in the request must not be present. Instead the request should look like this:
+ *
+ * <pre>
+ * PATCH /scim/Users/2752513
+ * {
+ *     "schemas": [
+ *         "urn:ietf:params:scim:api:messages:2.0:PatchOp"
+ *     ],
+ *     "Operations": [
+ *         {
+ *             "op": "replace",
+ *             "value": {
+ *                 "name": {
+ *                     "givenName": "captain",
+ *                     "familyName": "goldfish"
+ *                 }
+ *             }
+ *         }
+ *     ]
+ * }
+ * </pre>
+ */
+
+
+@Slf4j
+@RequiredArgsConstructor
+public final class MsAzurePatchReplaceWorkaroundHandler
+{
+
+  /**
+   * the patch operation that is currently executed
+   */
+  private final PatchOp patchOp;
+
+  /**
+   * the values of the patch operation. This attribute should actually be empty
+   */
+  private final List<String> values;
+
+
+  public List<String> fixValues()
+  {
+    // just a security check to make sure that the if-block that prevents this class to be executed in case of ADD
+    // and REMOVE should disappear
+    if (!PatchOp.REPLACE.equals(patchOp))
+    {
+      log.trace("[MS Azure REPLACE workaround] only handling 'REPLACE' requests");
+      return values;
+    }
+
+    // nothing must be done patch request can be handled normally since no illegal value operand is present
+    if (values.isEmpty())
+    {
+      log.trace("[MS Azure REPLACE workaround] workaround not executed for values-list is empty");
+      return values;
+    }
+
+    if (values.size() > 1)
+    {
+      log.trace("[MS Azure REPLACE workaround] workaround not executed for values-list with more than one value");
+      return values;
+    }
+
+    String value = values.get(0);
+
+    if (!JsonHelper.isValidJson(value))
+    {
+      // do nothing anymore this will cause the request to normally abort at the specific validation point
+      log.trace("[MS Azure REPLACE workaround] attribute in 'value' operand is not valid json: {}", value);
+      return values;
+    }
+
+    JsonNode jsonNode = JsonHelper.readJsonDocument(value);
+    final boolean isNodeAnObject = Optional.ofNullable(jsonNode).map(JsonNode::isObject).orElse(false);
+    if (!isNodeAnObject)
+    {
+      // do nothing anymore this will cause the request to normally abort at the specific validation point
+      log.trace("[MS Azure REPLACE workaround] attribute in 'value' operand is not an object: {}", value);
+      return values;
+    }
+    ObjectNode rootObjectNode = (ObjectNode)jsonNode;
+
+    Optional<List<String>> schemas = JsonHelper.getSimpleAttributeArray(rootObjectNode, AttributeNames.RFC7643.SCHEMAS);
+    ObjectNode resourceObjectNode = rootObjectNode;
+
+    // set extension resource object as resourceObjectNode if found
+    for ( Iterator<String> it = rootObjectNode.fieldNames() ; it.hasNext() ; )
+    {
+      String fieldName = it.next();
+
+      if (schemas.isPresent() && schemas.get().contains(fieldName))
```

Comment:
please move the `schemas.isPresent()` around the upper for-loop

Reply:
OK